components/properties/src/bidi.rs

```diff
@@ -0,0 +1,160 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! `unicode-bidi` is a crate that implements the [unicode bidi algorithm](https://unicode.org/reports/tr9/)
+//! by using the data from ICU4X through `BidiClassAdapter`
+//!
+//!```
+//! use icu_properties::bidi::BidiClassAdapter;
+//! use icu_properties::{maps, BidiClass};
+//! use icu_codepointtrie::CodePointTrie;
+//! use unicode_bidi::BidiClass as DataSourceBidiClass;
+//! use unicode_bidi::BidiDataSource;
+//! use unicode_bidi::BidiInfo;
+//! // This example text is defined using `concat!` because some browsers
+//! // and text editors have trouble displaying bidi strings.
+//! let text = concat![
```

Comment:
Question: Did you write this example yourself, or did you get it from somewhere else?

Reply:
I get it from the unicode-bidi crate.